### PR TITLE
Feat/ improve trip scheduler

### PR DESCRIPTION
**Pull request by Vincentdfvh**
This PR bounds the trip activities by the trip end day and also fixes a bug where the activities were not proposed for the last day of the trip.

- add documentation to further explain variables about locations in TripInfoScreen and TripScheduler
- add test to test the conditions to call next day : max activity overflow, activity time overflow and travel time overflow

### CODE REVIEW

File: app/src/main/java/com/github/swent/swisstravel/algorithm/selectactivities/SelectActivities.kt [hunk 1]
```diff
@@ -55,7 +55,10 @@ class SelectActivities(
     userPreferences.remove(Preference.EARLY_BIRD)
     userPreferences.remove(Preference.NIGHT_OWL)
 
-    val days = ChronoUnit.DAYS.between(tripSettings.date.startDate, tripSettings.date.endDate)
+    // add 1 day since the last day is excluded
+    val days =
+        ChronoUnit.DAYS.between(
+            tripSettings.date.startDate, tripSettings.date.endDate!!.plusDays(1))
```
[the remark this answers is not in the document]
Comment (Vincentdfvh) [reply app/src/main/java/com/github/swent/swisstravel/algorithm/selectactivities/SelectActivities.kt]: That's a good point. So, i just looked it up, and i don't there is another way to add the excluded day. I also did not find another function that includes the last day.
